### PR TITLE
[Android] Fixed issue changing the ItemTemplate on CarouselView

### LINKED ISSUE #7395: [Bug] CollectionView: changing ItemTemplate does not work as expected
### Description
If you change the ItemTemplate (programmatically) for a CollectionView, the template is only partially applied, or not at all. Some lines in the collection have the new template, others have the old template. Bij scrolling up and down, this changes, and sometimes in the end all lines have the new template. I made a simple app to reproduce the problem.
Only tested this on Android.
[CollectionViewTest.zip](https://github.com/xamarin/Xamarin.Forms/files/3578650/CollectionViewTest.zip)

### Steps to Reproduce

1. Take the attached sample. It starts with a CollectionView without template
2. Press "Three colums". Nothing happens. Scroll down and up a few times and notice the changes.
3. Press "Two columns", scroll and notice the changes.
4. In code, you can uncomment the lines that reset the ItemsSource. It maybe behaves slightly different, but not OK. Using an ObservableCollection makes no difference.

### Expected Behavior
The new template should be applied on all lines, immediately or maybe after some trigger (setting the ItemsSource or some other mechanism)
### Actual Behavior
Some lines accept the new template, others don't. Results change while scrolling
### Basic Information

- Version with issue: 4.2.0.709249
- Last known good version: ?
- IDE: Visual Studio 2019
- Platform Target Frameworks: 
  - Android: 8.1.0

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Fixed a bug changing the **ItemTemplate** dynamically. The change of the Template was not applied or partially when scrolling.

### Issues Resolved ### 

- fixes #7395

### API Changes ###
 
 None

### Platforms Affected ### 
- Android

### Behavioral/Visual Changes ###

None

#### Before
7395-before

#### After
7395-after

### Testing Procedure ###
Launch Core Gallery on Android and navigate to the Issue 7395. Press the buttons and see if the CollectionView ItemTemplate changes.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		{
+			_collectionView.ItemTemplate = CreateDataGridTemplate(2);
+		}
+
+		void Button2_Clicked(object sender, EventArgs e)
+		{
+			_collectionView.ItemTemplate = CreateDataGridTemplate(3);
+		}
+
+		private DataTemplate CreateDataGridTemplate(int columns)
```
Comment (adrianknight89) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs]: drop private

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+#if UITEST
+	[Category(UITestCategories.CollectionView)]
+#endif	
+ 	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 7395, "Changing ItemTemplate does not work as expected", PlatformAffected.Android)]
+	public class Issue7395 : TestContentPage
```
Comment (adrianknight89) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs]: Can you add instructions for future testers?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+			var stack = new StackLayout();
+			stack.Children.Add(button1);
+			stack.Children.Add(button2);
+			stack.Children.Add(_collectionView);
+
+			Content = stack;
+		}
+
+		void Button1_Clicked(object sender, EventArgs e)
```
Comment (rmarinho) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs]: We name methods without the underscore :)

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs (hunk 2 above)
Comment (jsuarezruiz) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs]: Sure!. Added instructions:
7395-instructions

Comment (adrianknight89) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs]: @jsuarezruiz Can we give instructions to click those buttons as well? 

"Click Two Columns. If all cells render correctly with 2 columns, then click Three Columns. If all cells render correctly with 3 columns, then the test passes."

Comment (jsuarezruiz) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs]: Sure, added.

Comment (adrianknight89) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7395.cs]: Thanks

**Pull request merged**